vhost-device-scmi/src/devices/iio.rs

```diff
@@ -286,6 +296,97 @@ impl SensorT for IIOSensor {
         }
         Ok(result)
     }
+
+    fn get_notify_fd(&self) -> Option<RawFd> {
+        self.sensor.notify_dev.as_ref().map(|fd| fd.as_raw_fd())
+    }
+
+    fn reading_device(&self, device_index: u32) -> DeviceResult {
+        let mut result = vec![];
+
+        //Here we should create buffer which length is corresponding to iio:device;
+        //future change to self.get_data_format
+        //cat /sys/bus/iio/devices/iio:deviceX/scan_elements/in_XXX_type.
+        //for example le:s16/16>>0 , [x,y,z,(t)] each is signed 16, and Little Endian
+        //so one sample length is (3 or 4)*(16/8) byte
```

Comment:
I'm not sure I understand it correctly, let me rephrase it and please correct it if it is wrong:

```
// The buffer length should correspond to the IIO device type.
// The type is available from /sys/bus/iio/devices/iio:deviceX/scan_elements/in_XXX_type.
// For example, if the content of in_XXX_type is le:s16/16>>0, each value is a little endian
// signed 16-bit integer. For a 3-axes sensor with [x, y, z, (t)] values, i.e. the 3 axes plus an
// optional timestamp, we need 6 or 8 bytes buffer.
```

Reply:
Perfect rephrase, exactly yes!

Reply:
Because rust-vmm backend is not clear about the package format of data from /dev/iio:deviceX.
So it's little difficult to re-package it and satisfy SCMI spec requirements.
Therefore, in our implement, we only simply delt like this and ignore timestamp.

In order to consider timestamp. The SENSOR_CONFIG_SET command should also be refactored.
According to scmi spec, the parameter "sensor_config" in command SENSOR_CONFIG_SET should stores more information. Maybe we should also add a variable into SensorT to store configuration of each sensor.

> Bit[1] Timestamp reporting:
> Set to 1 if the sensor value provided by the platform should be timestamped.
> Set to 0 if the sensor value provided by the platform should not be timestamped.
> If the sensor does not support timestamp reporting or its configuration, this bit should be set to 0.

Reply:
If timestamps are to be used then yes. Otherwise it's optional, since the current bit value 0 is right for non-timestamped values. Up to you.